OpenTrons/opentrons-api
# feat(g-code-testing): HTTP G-Code GitHub Action (RET-174)

# Overview

Add HTTP tests to Github action
Refactor framework to use pytest

# Changelog
- Add AWS python package, `boto3`, to `g-code-testing` project
- Revert passing classes to G-Code Engine and instead just pass parameters required for execution
- Remove GCodeTestData class and configurations file and any associated references/tests
- Remove cli and any associated references/tests
- Update HTTP and protocol G-Code confirmation tests to work with pytest
- Add pytest marks to tests to classify them
  - Add `slow` custom mark
  - Add `g_code_confirm` custom mark
- Update Makefile `test` targets
  - Classifications of tests
    1. Framework Unit Tests - tests to ensure that framework has not regressed - under normal `make test` call 
    2. Fast G-Code Confirm Tests - G-Code Confirmation tests that run quickly (sub 1 minute) - under `test-g-code-fast`
    3. Slow G-Code Confirm Tests - G-Code Confirmation tests that run quickly (above 1)  - separated out into specific targets per test to allow for parallelization in Github Action. For example, `make test-g-code-swift-smoke`
- Consolidate `g-code-testing` project into 2 actions
  - `G-Code Testing Lint & Test` - Run linting and `Framwork Unit Tests`
  - `G-Code-Confirm` - Run `Fast G-Code Confirm Tests` in single job and each slow test in it's own job (Each job runs in parallel)



# Review requests

Take a look at the README and note the Control Tests section. Does it make sense to skip the tests that I noted as skipped?

# Risk assessment

Could break G-Code Confirmation tests. Rerunning tests in CI should take care of this though
